**Importando as bibliotecas necessárias e visão geral do dataset**

In [ ]:
# Bibliotecas para ajudar na leitura e manipulação de dados
import pandas as pd
import matplotlib.pyplot as plt

# Bibliotecas para ajudar na visualização de dados
import seaborn as sns

**Carregando o conjunto de dados**

In [ ]:
import pandas as pd

df_chamados = pd.read_csv("chamados.csv")

**Visualizando as 5 primeiras linhas do conjunto de dados**

In [ ]:
df_chamados.head()

,ticket_id,created_at,customer_id,department,city_x,state_x,device_type,os,browser,network_type,...,vip,monthly_revenue,num_events,last_event_date,event_types,num_abertura,num_updates,num_encerramento,num_images_y,target_category
0,1,2025-01-01T00:17:00,5012,Operações,Campinas,BA,Desktop,Windows,Safari,WiFi,...,NaN,NaN,0,NaN,NaN,0,0,0,NaN,Hardware
1,2,2025-01-01T00:34:00,6574,Financeiro,Campinas,RJ,Notebook,Windows,Chrome,5G,...,NaN,NaN,0,NaN,NaN,0,0,0,NaN,RH
2,3,2025-01-01T00:51:00,2139,TI,Campinas,MG,Desktop,Linux,Chrome,5G,...,NaN,NaN,0,NaN,NaN,0,0,0,NaN,Financeiro
3,4,2025-01-01T01:08:00,7216,Financeiro,Belo Horizonte,RJ,Mobile,Android,Chrome,Cabo,...,NaN,NaN,0,NaN,NaN,0,0,0,NaN,Software
4,5,2025-01-01T01:25:00,5040,Financeiro,Belo Horizonte,PR,Mobile,iOS,Edge,Cabo,...,NaN,NaN,0,NaN,NaN,0,0,0,NaN,Rede


**Verificando a forma (shape) do conjunto de dados**

In [ ]:
df_chamados.shape

(5000, 44)



*   O conjunto de dados possui 5.000 linhas e 44 colunas.







**Verificando o info()**

In [ ]:
df_chamados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 44 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   ticket_id              5000 non-null   int64  
 1   created_at             5000 non-null   object 
 2   customer_id            5000 non-null   int64  
 3   department             5000 non-null   object 
 4   city_x                 5000 non-null   object 
 5   state_x                5000 non-null   object 
 6   device_type            5000 non-null   object 
 7   os                     5000 non-null   object 
 8   browser                5000 non-null   object 
 9   network_type           5000 non-null   object 
 10  historical_tickets     5000 non-null   int64  
 11  sla_hours              5000 non-null   int64  
 12  resolution_time_hours  5000 non-null   int64  
 13  title                  5000 non-null   object 
 14  description            5000 non-null   object 
 15  imag

**Observações**


*   As colunas estão divididas entre variáveis numéricas (int64 e float64) e categóricas/textuais (object).

*   As principais informações dos chamados, como ticket_id, created_at, department, priority, status, assigned_team, resolution_time_hours e target_category, possuem 5.000 valores preenchidos, não apresentando valores ausentes.

*   As variáveis title e description são textuais e podem ser utilizadas posteriormente em análises de texto ou processamento de linguagem natural.

*   As variáveis sentiment_score, num_words, historical_tickets, sla_hours e resolution_time_hours são numéricas e podem ser utilizadas para análises estatísticas.

*   As colunas company, segment, employees, city_y, state_y, contract_type, customer_since, vip e monthly_revenue possuem apenas 2 registros preenchidos de 5.000, indicando uma quantidade extremamente elevada de valores ausentes.

*   As colunas last_event_date, event_types e num_images_y possuem 0 valores preenchidos, ou seja, estão completamente vazias.

*   As colunas relacionadas a eventos, como num_events, num_abertura, num_updates e num_encerramento, possuem dados completos para os 5.000 registros.

*   A variável target_category possui 5.000 registros preenchidos e aparenta ser uma variável importante para o objetivo de classificação do projeto.








**Verificar valores ausentes**

In [ ]:
df_chamados.isnull().sum().sort_values(ascending=False)

,0
last_event_date,5000
event_types,5000
num_images_y,5000
monthly_revenue,4998
contract_type,4998
employees,4998
city_y,4998
segment,4998
company,4998
vip,4998


In [ ]:
missing = pd.DataFrame({
    'quantidade': df_chamados.isnull().sum(),
    'percentual': df_chamados.isnull().mean() * 100
})

missing.sort_values('percentual', ascending=False)

,quantidade,percentual
last_event_date,5000,100.00
event_types,5000,100.00
num_images_y,5000,100.00
monthly_revenue,4998,99.96
contract_type,4998,99.96
employees,4998,99.96
city_y,4998,99.96
segment,4998,99.96
company,4998,99.96
vip,4998,99.96


**Verificar valores duplicados**

In [ ]:
df_chamados.duplicated().sum()

np.int64(0)

In [ ]:
df_chamados['ticket_id'].duplicated().sum()

np.int64(0)

In [ ]:
df_chamados.dtypes

,0
ticket_id,int64
created_at,object
customer_id,int64
department,object
city_x,object
state_x,object
device_type,object
os,object
browser,object
network_type,object


**Conversão das variáveis de data**

In [ ]:
df_chamados['created_at'] = pd.to_datetime(
    df_chamados['created_at'],
    errors='coerce'
)

In [ ]:
df_chamados['created_at'].dtype

dtype('<M8[ns]')

In [ ]:
df_chamados['created_at'].isna().sum()

np.int64(0)

In [ ]:
errors='coerce'

In [43]:
print("PRIORIDADE:")
print(df_chamados['priority'].value_counts())

print("\nSTATUS:")
print(df_chamados['status'].value_counts())

print("\nDEPARTAMENTO:")
print(df_chamados['department'].value_counts())

print("\nCATEGORIA:")
print(df_chamados['target_category'].value_counts())

print("\ncity_x:")
print(df_chamados['city_x'].value_counts())

print("\ncity_y:")
print(df_chamados['city_y'].value_counts())

print("\nstate_y:")
print(df_chamados['state_y'].value_counts())

print("\nstate_x:")
print(df_chamados['state_x'].value_counts())

print("\nos:")
print(df_chamados['os'].value_counts())

print("\nbrowser:")
print(df_chamados['browser'].value_counts())

print("\ndevice_type:")
print(df_chamados['device_type'].value_counts())

print("\nnetwork_type:")
print(df_chamados['network_type'].value_counts())






PRIORIDADE:
priority
Crítica    1277
Baixa      1260
Média      1248
Alta       1215
Name: count, dtype: int64

STATUS:
status
Aberto          1689
Fechado         1657
Em andamento    1654
Name: count, dtype: int64

DEPARTAMENTO:
department
RH            1274
TI            1273
Operações     1229
Financeiro    1224
Name: count, dtype: int64

CATEGORIA:
target_category
Software      892
Rede          861
Hardware      835
RH            821
Acesso        821
Financeiro    770
Name: count, dtype: int64

city_x:
city_x
Santos            1012
Belo Horizonte    1008
Campinas          1002
Curitiba          1000
São Paulo          978
Name: count, dtype: int64

city_y:
city_y
Curitiba     1
São Paulo    1
Name: count, dtype: int64

state_y:
state_y
PR    1
SP    1
Name: count, dtype: int64

state_x:
state_x
RS    862
SP    849
BA    837
MG    834
RJ    822
PR    796
Name: count, dtype: int64

os:
os
Windows    1256
Android    1252
Linux      1248
iOS        1244
Name: count, dtype: int64

br

**Observações — variáveis categóricas**

A análise das variáveis categóricas priority, status, department e target_category não identificou inconsistências aparentes na nomenclatura das categorias. Os valores apresentam padronização de escrita e não foram observadas categorias duplicadas por diferenças de maiúsculas, minúsculas ou grafia.



Também podemos registrar os resultados:

*   Prioridade: Crítica (1.277), Baixa (1.260), Média (1.248) e Alta (1.215).

*   Status: Aberto (1.689), Fechado (1.657) e Em andamento (1.654).

*   Departamento: RH (1.274), TI (1.273), Operações (1.229) e Financeiro (1.224).

*   Categoria: Software (892), Rede (861), Hardware (835), RH (821), Acesso (821) e Financeiro (770).




**Análise estatística das variáveis numéricas**

In [ ]:
df_chamados.describe()

,ticket_id,created_at,customer_id,historical_tickets,sla_hours,resolution_time_hours,image_width,image_height,has_attachment,sentiment_score,...,longitude,employees,monthly_revenue,num_events,last_event_date,event_types,num_abertura,num_updates,num_encerramento,num_images_y
count,5000.000000,5000,5000.000000,5000.000000,5000.000000,5000.00000,5000.00000,5000.000000,5000.000000,5000.00000,...,5000.000000,2.000000,2.000000,5000.0,0.0,0.0,5000.0,5000.0,5000.0,0.0
mean,2500.500000,2025-01-30 12:28:30,5517.150600,19.933600,19.135200,36.02900,553.29060,371.952000,0.499200,-0.00319,...,-48.097434,217.500000,102500.000000,0.0,NaN,NaN,0.0,0.0,0.0,NaN
min,1.000000,2025-01-01 00:17:00,1006.000000,0.000000,4.000000,1.00000,191.00000,165.000000,0.000000,-1.00000,...,-53.099397,85.000000,25000.000000,0.0,NaN,NaN,0.0,0.0,0.0,NaN
25%,1250.750000,2025-01-15 18:22:45,3171.000000,10.000000,8.000000,18.00000,500.00000,333.000000,0.000000,-0.51625,...,-50.514807,151.250000,63750.000000,0.0,NaN,NaN,0.0,0.0,0.0,NaN
50%,2500.500000,2025-01-30 12:28:30,5540.000000,20.000000,12.000000,36.00000,500.00000,342.000000,0.000000,0.00500,...,-48.104285,217.500000,102500.000000,0.0,NaN,NaN,0.0,0.0,0.0,NaN
75%,3750.250000,2025-02-14 06:34:15,7813.250000,30.000000,24.000000,54.00000,612.00000,408.000000,1.000000,0.50600,...,-45.638371,283.750000,141250.000000,0.0,NaN,NaN,0.0,0.0,0.0,NaN
max,5000.000000,2025-03-01 00:40:00,9999.000000,40.000000,48.000000,72.00000,738.00000,750.000000,1.000000,1.00000,...,-43.201069,350.000000,180000.000000,0.0,NaN,NaN,0.0,0.0,0.0,NaN
std,1443.520003,NaN,2627.870259,11.827813,15.809355,20.82298,111.36053,79.284335,0.500049,0.58097,...,2.857408,187.383297,109601.551084,0.0,NaN,NaN,0.0,0.0,0.0,NaN


**Observações — análise estatística**

A análise estatística das variáveis numéricas mostrou que as variáveis historical_tickets, sla_hours e resolution_time_hours apresentam valores dentro de faixas definidas no conjunto de dados. O tempo médio de resolução dos chamados é de aproximadamente 36 horas, enquanto o SLA médio é de aproximadamente 19 horas, indicando a necessidade de investigar o cumprimento dos SLAs. A variável sentiment_score apresenta valores entre -1 e 1, com média próxima de zero. A variável has_attachment apresenta valores binários entre 0 e 1, com aproximadamente metade dos registros indicando a presença de anexos.

Também foi identificado que as variáveis num_events, num_abertura, num_updates e num_encerramento apresentam valor zero em todos os registros, sendo necessária uma investigação antes de utilizá-las ou removê-las. As variáveis employees e monthly_revenue possuem apenas dois valores válidos, enquanto last_event_date, event_types e num_images_y não possuem valores preenchidos.

**Análise do cumprimento do SLA**

In [ ]:
df_chamados['sla_estourado'] = (
    df_chamados['resolution_time_hours'] >
    df_chamados['sla_hours']
)

In [ ]:
df_chamados['sla_estourado'].value_counts()

,count
sla_estourado,
True,3661
False,1339


In [ ]:
df_chamados['sla_estourado'].value_counts(normalize=True) * 100

,proportion
sla_estourado,
True,73.22
False,26.78


**Análise do SLA por prioridade**

In [ ]:
sla_prioridade = df_chamados.groupby('priority')['sla_estourado'].agg(
    quantidade='count',
    sla_estourado='sum',
    percentual_estourado='mean'
)

sla_prioridade['percentual_estourado'] *= 100

sla_prioridade

,quantidade,sla_estourado,percentual_estourado
priority,,,
Alta,1215,887,73.004115
Baixa,1260,921,73.095238
Crítica,1277,932,72.983555
Média,1248,921,73.798077


A análise do cumprimento do SLA por prioridade mostrou que os percentuais de chamados com SLA estourado são bastante semelhantes entre as categorias. A prioridade Média apresentou o maior percentual, com 73,80%, enquanto a prioridade Crítica apresentou o menor, com 72,98%. A pequena diferença entre as categorias indica que, neste conjunto de dados, a prioridade do chamado não parece produzir uma diferença significativa no cumprimento do SLA.

**Análise do SLA por departamento**

In [ ]:
sla_departamento = df_chamados.groupby('department')['sla_estourado'].agg(
    quantidade='count',
    sla_estourado='sum',
    percentual_estourado='mean'
)

sla_departamento['percentual_estourado'] *= 100

sla_departamento

,quantidade,sla_estourado,percentual_estourado
department,,,
Financeiro,1224,900,73.529412
Operações,1229,896,72.904801
RH,1274,929,72.919937
TI,1273,936,73.527101


A análise do cumprimento do SLA por departamento apresentou resultados bastante semelhantes entre os setores. Financeiro e TI apresentaram as maiores taxas de SLA estourado, ambas de aproximadamente 73,53%, enquanto Operações apresentou a menor taxa, com 72,90%. A pequena diferença entre os departamentos indica que o setor responsável pelo atendimento não apresenta grande variação na taxa de cumprimento do SLA neste conjunto de dados.

**Análise do SLA por categoria**

In [ ]:
sla_categoria = df_chamados.groupby('target_category')['sla_estourado'].agg(
    quantidade='count',
    sla_estourado='sum',
    percentual_estourado='mean'
)

sla_categoria['percentual_estourado'] *= 100

sla_categoria.sort_values('percentual_estourado', ascending=False)

,quantidade,sla_estourado,percentual_estourado
target_category,,,
Hardware,835,623,74.610778
Software,892,665,74.551570
Rede,861,631,73.286876
RH,821,598,72.838002
Financeiro,770,560,72.727273
Acesso,821,584,71.132765


A análise do cumprimento do SLA por categoria apresentou diferenças entre os tipos de chamados. Hardware apresentou a maior proporção de chamados com SLA estourado, com 74,61%, seguido por Software, com 74,55%. A categoria Acesso apresentou a menor proporção, com 71,13%. A diferença de aproximadamente 3,48 pontos percentuais entre a maior e a menor taxa indica uma variação moderada no cumprimento do SLA entre as categorias.